### PR TITLE
Fix various issues for compatibility with Linguist

### PR DESCRIPTION
**Pull request by am11**
First wave of errors from linguist compiler look like this:

```
script/grammar-compiler add vendor/grammars/Nixinova-tmLanguage
  > latest: Pulling from linguist/grammar-compiler
  > Digest: sha256:022a1c69f9bbe4edb1b2c6c1476266c6e0ef5395abfd39f98e1e676d055f98f2
  > Status: Image is up to date for linguist/grammar-compiler:latest
  > docker.io/linguist/grammar-compiler:latest
  > The new grammar repository `vendor/grammars/Nixinova-tmLanguage` (from https://github.com/am11/tmLanguage) contains 2 errors:
  >     - Grammar conversion failed. File `grammars/robots-txt.yaml-tmLanguage` failed to parse: 1 error(s) decoding:
  >
  > * 'Repository[comment].BeginCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  >     - Grammar conversion failed. File `grammars/solution.yaml-tmLanguage` failed to parse: 13 error(s) decoding:
  >
  > * 'Repository[functions].Patterns[0].BeginCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[functions].Patterns[0].EndCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[functions].Patterns[1].BeginCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[functions].Patterns[1].EndCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[header].Patterns[1].BeginCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[strings].Patterns[0].BeginCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[strings].Patterns[0].EndCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[strings].Patterns[1].BeginCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[strings].Patterns[1].EndCaptures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[variables].Patterns[0].Captures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[variables].Patterns[0].Captures[%!s(int=1)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[variables].Patterns[1].Captures[%!s(int=0)]' expected type 'string', got unconvertible type 'int'
  > * 'Repository[variables].Patterns[1].Captures[%!s(int=1)]' expected type 'string', got unconvertible type 'int'
  >
  > failed to compile the given grammar
```

then it complains about a lookbehind regex pattern: ` 'lookbehind assertion is not fixed length'` due to asterisk in `\s*`. I have simplified it by removing lookbehind.

### CODE REVIEW

**Comment by Nixinova** [comment]
Are letters valid aliases for capture groups?

File: grammars/solution.yaml-tmLanguage [hunk 1]
```diff
@@ -54,15 +54,15 @@ repository:
...
       - name: meta.generic.variable.property.solution
         match: '([A-Za-z]+)\s*(\|)'
         captures:
-          0: { name: variable.language.property.solution }
-          1: { name: punctuation.separator.property.solution }
+          a: { name: variable.language.property.solution }
+          b: { name: punctuation.separator.property.solution }
       - name: variable.other.member.solution
-        match: '(?<=[.|]\s*)[A-Za-z0-9 ]+'
+        match: '[.|]\s*[A-Za-z0-9 ]+'
```
Comment (Nixinova) [inline grammars/solution.yaml-tmLanguage:65]: This should still be a look behind, just move the \s* outside.

Comment (am11) [reply grammars/solution.yaml-tmLanguage]: Ok. I had it with lookbehind with \s* outside earlier, but then thought LB is not necessary. Is there an example where it is needed?

**Comment by am11** [comment]
> Are letters valid aliases for capture groups?

Only quotes were needed around the numbers to satisfy PCRE lexical rules.

File: grammars/solution.yaml-tmLanguage (hunk 1 above)
Comment (Nixinova) [reply grammars/solution.yaml-tmLanguage]: It's just to exclude the "|"/"." from being included in the match, as it should be marked as just punctuation and not be part of the member-variable token.

**Review by Nixinova — APPROVED** [review]

**Comment by Nixinova** [comment]
Main compatibility issue (the generic repo name) is being fixed with https://github.com/github/linguist/pull/5292

**Comment by am11** [comment]
> Main compatibility issue

So you are saying the 10+ fatal errors from linguist compiler I posted above are not the main issue? i.e. without this change you can fix it? Please show us how..

**Comment by Nixinova** [comment]
I mean the main outstanding issue

**Pull request merged**